### PR TITLE
Show decorations in the editor tabs (#13301)

### PR DESCRIPTION
**Pull request by dhuebner**
#### What it does

As described in  #13301 there were no git status indication in the editor tabs.  Now all decorations contributed to the `DecorationsService` are rendered in tabs that owned by a `Navigatable` Widget.  This currently includes SCM (git) and Problem decorations.

Rendering is controlled with `workbench.editor.decorations.badges`  preference.

#### How to test

Open en example workspace, produce some SCM changes and errors. You should see tail decoration appearing in the editor tabs similar to the navigator.

Bildschirmfoto 2024-02-09 um 14 04 13

#### Review checklist

- [ x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/browser/shell/tab-bar-decorator.ts [hunk 1]
```diff
@@ -71,6 +80,27 @@ export class TabBarDecoratorService implements FrontendApplicationContribution {
             const decorations = decorator.decorate(title);
             all = all.concat(decorations);
```
[the remark this answers is not in the document]
Comment (dhuebner) [reply packages/core/src/browser/shell/tab-bar-decorator.ts:81]: Just did the same as guys before to match the style. But yes, we can change that.

File: packages/core/src/browser/shell/tab-bar-decorator.ts [hunk 2]
```diff
@@ -71,6 +80,27 @@ export class TabBarDecoratorService implements FrontendApplicationContribution {
...
+        if (Navigatable.is(title.owner)) {
+            if (title.owner.getResourceUri() !== undefined) {
+                const serviceDecorations = this.decorationsService.getDecoration(title.owner.getResourceUri()!, false);
+                all = all.concat(serviceDecorations.map(d => this.toDecorator(d)));
+            }
+        }
         return all;
     }
+
+    protected toDecorator(decoration: Decoration): WidgetDecoration.Data {
```
[the remark this answers is not in the document]
Comment (dhuebner) [reply packages/core/src/browser/shell/tab-bar-decorator.ts]: I see. Changed to `fromDecoration` now.

File: packages/core/src/browser/shell/tab-bar-decorator.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhuebner) [reply packages/core/src/browser/shell/tab-bar-decorator.ts]: Changed now.

File: packages/core/src/browser/shell/tab-bars.ts [hunk 3]
```diff
@@ -289,6 +290,38 @@ export class TabBarRenderer extends TabBar.Renderer {
         return h.div({ className: 'p-TabBar-tabLabel', style }, data.title.label);
     }
 
+    protected renderTailDecorations(renderData: SideBarRenderData, isInSidePanel?: boolean): VirtualElement[] {
+        if (!this.corePreferences?.get('workbench.editor.decorations.badges')) {
+            return [];
+        }
+        const tailDecorations = this.getDecorationData(renderData.title, 'tailDecorations')
+            .filter(acc => acc !== undefined).reduce((acc, current) => acc!.concat(current!), []);
```
[the remark this answers is not in the document]
Comment (dhuebner) [reply packages/core/src/browser/shell/tab-bars.ts]: Using the utility now.

**Comment by dhuebner** [comment]
@msujew 
Thanks for the re-view. All suggestion were applied.

File: packages/core/src/browser/shell/tab-bars.ts [hunk 4]
```diff
@@ -289,6 +290,38 @@ export class TabBarRenderer extends TabBar.Renderer {
         return h.div({ className: 'p-TabBar-tabLabel', style }, data.title.label);
     }
 
+    protected renderTailDecorations(renderData: SideBarRenderData, isInSidePanel?: boolean): VirtualElement[] {
+        if (!this.corePreferences?.get('workbench.editor.decorations.badges')) {
+            return [];
+        }
+        const tailDecorations = ArrayUtils.coalesce(this.getDecorationData(renderData.title, 'tailDecorations'))
+            .reduce((acc, current) => acc.concat(current), []);
```
Comment (msujew) [inline packages/core/src/browser/shell/tab-bars.ts:298]: Minor:
```suggestion
        const tailDecorations = ArrayUtils.coalesce(this.getDecorationData(renderData.title, 'tailDecorations')).flat();
```

**Review by msujew — APPROVED** [review]
Looks good to me. Just a minor issue, see below 👍

**Comment by dhuebner** [comment]
@msujew 
I've adjusted the code like suggested. Thank you for the re-view!

**Pull request merged**